src/dmd/link.d

```diff
@@ -562,59 +562,148 @@ public int runLINK()
             argv.push("-Xlinker");
             argv.push("--gc-sections");
         }
+
+        /// Dynamic array of strings to temporarily hold accumulated,
+        /// reordered linker args
+        Strings linkerArgs;
+
+        /**
+        Checks if a C string starts with a given string
+        Params:
+            p = the C string to check
+            needle = the string to look for
+        Returns
+            `true` if `p` starts with `needle`
+        */
+        static bool startsWith(const(char)* p, string needle)
+        {
+            const f = p[0 .. strlen(p)];
+            return f.length >= needle.length && f[0 .. needle.length] == needle;
+        }
+
+        /**
+        Checks if a C string is null or an emtpy string
+        Params:
+            p = the C string to check
+        Returns:
+            `true` if `p` is null or an empty string
+        */
+        static bool isNullOrEmpty(const(char)* p)
```

Comment:
I don't think this function adds any value, it's only used twice.

Reply:
Addressed in https://github.com/dlang/dmd/pull/10341